### ISSUE TITLE
queue.max_bytes isn't compared against queue.page_capacity

### ISSUE THREAD

**Issue by jsvd** [elastic/logstash#6627]
setup:

```
queue.type: persisted
queue.page_capacity: 250mb
queue.max_bytes: 10mb
```
with pipeline: `input { generator {} } filter {sleep { time => 1 } }` and `-w 1 -b 1`

Starting logstash will cause it to print 1 event and stop:

```
/tmp/logstash-5.1.2 % bin/logstash -e "input { generator {} } filter {sleep { time => 1 } }" -w 1 -b 1
{
      "sequence" => 0,
    "@timestamp" => 2017-02-01T21:14:26.883Z,
      "@version" => "1",
          "host" => "Joaos-MBP-5.lan",
       "message" => "Hello world!"
}
# pipeline stalls
```

Data created:

```
/tmp/logstash-5.1.2 % du -sh data/**/*
250M	data/queue
4.0K	data/queue/checkpoint.head
250M	data/queue/page.0
4.0K	data/uuid
```

**Comment by colinsurprenant** [comment]
good catch.

**Comment by jsvd** [comment]
this can be another bootstrap check done in the context of https://github.com/elastic/logstash/issues/6552

**Comment by colinsurprenant** [comment]
agree, will be fixed by #6998

**Comment by colinsurprenant** [comment]
fixed by #8978
